### DISCUSSION TITLE
undefined reference to `core::panicking::panic'

### DISCUSSION THREAD

**Discussion by kpreid** [pd-rs/crankstart#66] [Q&A] [answered]
I'm trying to use `crank` and `crankstart` as directed in the READMEs. However, building for the device fails (while the simulator succeeds):

```
$ crank build --device
   Compiling compiler_builtins v0.1.100
...
/usr/local/playdate/gcc-arm-none-eabi-9-2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld: <path …>: in function `compiler_builtins::mem::impls::copy_forward::copy_forward_misaligned_words':
/<path…>/compiler_builtins-0.1.100/src/mem/impls.rs:76: undefined reference to `core::panicking::panic'
collect2: error: ld returned 1 exit status
```

I'm currently trying to just compile my own copy-and-paste of the `hello_world` example. Does this indicate something I need to fix, or configure, or is it a bug in the tool?

The only previous mention I saw of this error is <https://github.com/pd-rs/crank/pull/42#issuecomment-1627564814>.

* OS: macOS
* Playdate SDK: 2.0.3
* `crank`: crankstart-cli v0.2.9 (rev = 155da63746a1408f6cdd2e120bee64c52871c58b)

**Comment by boozook** [comment]
Hey @kpreid 👋
Is there rust-src rustup component installed?

**Reply by kpreid** [reply]
Yes:

```
$ rustup +nightly component add rust-src
info: component 'rust-src' is up to date
```

**Reply by boozook** [reply]
I can't reproduce unfortunately.
But I have an idea that can help me determine what's going wrong.

Could you try run on device any example in [that][playdate] project?

Just because I know differences between two projects ;)

[playdate]: https://github.com/boozook/playdate

**Reply by parasyte** [reply, marked as answer]
I see that as well for debug builds. But it builds fine with release mode.

```
$ crank build --device --release
```

Now that I got my hardware back from RMA, this is the first time I've tried my Rust project on it. And it boots to a pure white screen. That's disappointing.

The `sprite_game` example works correctly on the hardware, though! So, this white screen thing is my own bug.

**Reply by kpreid** [reply]
Ah, the `build` or `run` worked with `--release`. I guess that makes this a bug with a workaround.

**Reply by boozook** [reply]
> Now that I got my hardware back from RMA, this is the first time I've tried my Rust project on it.

I'm glad to hear that, congrats!
And thanks for help. I didn't recognise that 🤦🏻‍♂️

**Reply by parasyte** [reply]
> I guess that makes this a bug with a workaround.

Yep! I would like to see the bug fixed as well. So, uh, I did it: https://github.com/pd-rs/crank/pull/52

**Reply by boozook** [reply]
Great! Merged. ❤️‍🔥
_I have to say that cargo-playdate supports debug builds as well as probably any custom profile ;)_

**Comment by boozook** [comment]
Summon @adamsoutar, @parasyte,
Do you have ideas for this?
